Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/Identity.cs

```diff
@@ -129,4 +129,18 @@ private IEnumerable<Quota> GetAppliedQuotasForMetric(MetricKey metric)
         var appliedQuotas = allQuotasOfMetric.Where(q => q.Weight == highestWeight).ToArray();
         return appliedQuotas;
     }
+
+    public async Task ChangeTierAsync(Tier newTier, MetricCalculatorFactory metricCalculatorFactory, CancellationToken cancellationToken)
+    {
```

Comment:
To make sure you don't do any unnecessary work, you should check if the newTier actually differs from the current one and return if they are the same.

Reply:
If they are the same, should the method simply return, or should we thrown an exception?

Reply:
I think we should throw. Just to make sure the caller notices that there was something wrong. It may be that they selected the wrong tier by mistake.

Reply:
It won't get to the domain code because we're checking in the handler before. I was thinking more about future code using this function.

Reply:
You should remove the check in the handler. Always put as much logic as possible into Domain